Sources/ScriptingCore/Models.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import TSCBasic
+import TSCUtility
+
+public struct PackageModel: Codable {
+    public let raw: String
+    public let url: Foundation.URL?
+    public let path: AbsolutePath?
+    private let _name: String?
+
+    public var name: String {
+        if let name = _name {
+            return name
+        }
+        let name: String
+        if let path = path {
+            name = path.basename
+        } else if let url = url {
+            name = url.pathComponents.last!.spm_dropGitSuffix()
+        } else {
+            fatalError()
```

Comment:
It would be great to avoid this in the code, but rather throw an error if possible.  I wonder if this `name` property should be computed in this way or if there should be a more explicit function that is clear that it is doing a bunch of checking and that can maybe throw an error.